test/sql/test_default_value/T/test_json_default.sql

```diff
@@ -0,0 +1,524 @@
+DISTRIBUTED BY HASH(id) BUCKETS 1
+PROPERTIES("replication_num" = "1");
+
+INSERT INTO with_default (id) VALUES (1);
+INSERT INTO without_default (id) VALUES (1);
+
+-- Compare: default value vs NULL
+SELECT 
+    'with_default' AS table_name,
+    data,
+    data IS NULL AS is_null
+FROM with_default
+UNION ALL
+SELECT 
+    'without_default',
+    data,
+    data IS NULL
+FROM without_default;
+
+-- Test: Empty string handling
+CREATE TABLE empty_string_test (
+    id INT,
+    empty_unquoted JSON DEFAULT '',
+    empty_quoted JSON DEFAULT '""'
+) DUPLICATE KEY(id)
+DISTRIBUTED BY HASH(id) BUCKETS 1
+PROPERTIES("replication_num" = "1");
+
+INSERT INTO empty_string_test (id) VALUES (1);
+INSERT INTO empty_string_test VALUES (2, '', '""');
+
+SELECT 
+    id,
+    empty_unquoted,
+    empty_quoted,
+    empty_unquoted = empty_quoted AS are_equal
+FROM empty_string_test
+ORDER BY id;
+
+-- ====================================================================================
+-- SECTION 2: ALTER TABLE ADD COLUMN - Fast Schema Evolution
+-- ====================================================================================
+
+-- Test: Add JSON column to existing table (fast schema evolution)
+CREATE TABLE fast_schema_change (
+    id INT NOT NULL,
+    name VARCHAR(50)
+) DUPLICATE KEY(id)
+DISTRIBUTED BY HASH(id) BUCKETS 1
+PROPERTIES("replication_num" = "1", "fast_schema_evolution" = "true");
+
+-- Insert data before adding JSON column
+INSERT INTO fast_schema_change VALUES (1, 'alice'), (2, 'bob');
+
+-- Add JSON column with default value
+ALTER TABLE fast_schema_change ADD COLUMN metadata JSON DEFAULT '{"version": 1, "enabled": true}';
+
+-- Old rows should use default value
+SELECT id, name, metadata FROM fast_schema_change ORDER BY id;
+
+-- New rows can have actual values
+INSERT INTO fast_schema_change VALUES (3, 'charlie', '{"version": 2, "enabled": false}');
+
+SELECT id, name, metadata FROM fast_schema_change ORDER BY id;
+
+-- ====================================================================================
+-- SECTION 3: ALTER TABLE ADD COLUMN - Traditional Schema Change
+-- ====================================================================================
+
+-- Test: Add JSON column with data rewrite (traditional schema change)
+CREATE TABLE traditional_schema_change (
+    id INT NOT NULL,
+    value INT
+) DUPLICATE KEY(id)
+DISTRIBUTED BY HASH(id) BUCKETS 1
+PROPERTIES("replication_num" = "1", "fast_schema_evolution" = "false");
+
+INSERT INTO traditional_schema_change VALUES (1, 100), (2, 200);
+
+-- Add JSON column - triggers full data rewrite
+ALTER TABLE traditional_schema_change ADD COLUMN metadata JSON DEFAULT '{"source": "migration", "timestamp": 0}';
+
+function: wait_alter_table_finish()
+
+SELECT count(*) FROM traditional_schema_change;
+
+-- ====================================================================================
+-- SECTION 4: Extended Column with JSON Functions (FlatJSON Optimization)
```

Comment:
why do you need to test extended column with default json value ? does it deserve a separated test case ?